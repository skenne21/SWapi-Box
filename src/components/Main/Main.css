.main {
  width: 80%;
  background-color: black;
}