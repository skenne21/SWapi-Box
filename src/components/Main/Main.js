import React from 'react';
import PropTypes from 'prop-types';
import ScrollText from '../ScrollText/ScrollText.js';
import CardsContainer from '../CardsContainer/CardsContainer.js';
import './Main.css'

const Main = ({film, cards}) => {
  return (
    <main> 
    {
      cards.length ? 
      <CardsContainer cards={cards} />
      :
      <ScrollText film={film} />
    }
    </main>
  
  )
}

Main.propTypes = {
  film: PropTypes.object,
  cards: PropTypes.array.isRequired
}

export default Main;

