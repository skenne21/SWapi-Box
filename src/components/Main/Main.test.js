import React from 'react';
import {shallow} from 'enzyme';
import Main from './Main.js';

describe('Main', () => {
  it('Should match the snapshot', () => {
    const film = {
      title: 'Attack of the clones',
      opening: 'It is a dark time for...',
      year: '2002'
    }
    const wrapper = shallow(<Main film={film} />)
    expect(wrapper).toMatchSnapshot();
  })
})