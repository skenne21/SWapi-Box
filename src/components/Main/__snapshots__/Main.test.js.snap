// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Should  create a Scroll Text if the CardsArray is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    cards={Array []}
    film={
        Object {
            "opening": "It is a dark time for...",
            "title": "Attack of the clones",
            "year": "2002",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ScrollText
        film={
                Object {
                        "opening": "It is a dark time for...",
                        "title": "Attack of the clones",
                        "year": "2002",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "film": Object {
          "opening": "It is a dark time for...",
          "title": "Attack of the clones",
          "year": "2002",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ScrollText
          film={
                    Object {
                              "opening": "It is a dark time for...",
                              "title": "Attack of the clones",
                              "year": "2002",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "film": Object {
            "opening": "It is a dark time for...",
            "title": "Attack of the clones",
            "year": "2002",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main Should create a cardsContainer component if there are items in the Cards array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    cards={
        Array [
            Object {
              "homeWorldName": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CardsContainer
        cards={
                Array [
                        Object {
                          "homeWorldName": "Tatooine",
                          "name": "Luke Skywalker",
                          "population": "200000",
                          "species": "Human",
                        },
                      ]
        }
        favorites={undefined}
        showFavorites={undefined}
        targetFavorites={undefined}
        toggleFavorites={undefined}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "cards": Array [
          Object {
            "homeWorldName": "Tatooine",
            "name": "Luke Skywalker",
            "population": "200000",
            "species": "Human",
          },
        ],
        "favorites": undefined,
        "showFavorites": undefined,
        "targetFavorites": undefined,
        "toggleFavorites": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CardsContainer
          cards={
                    Array [
                              Object {
                                "homeWorldName": "Tatooine",
                                "name": "Luke Skywalker",
                                "population": "200000",
                                "species": "Human",
                              },
                            ]
          }
          favorites={undefined}
          showFavorites={undefined}
          targetFavorites={undefined}
          toggleFavorites={undefined}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cards": Array [
            Object {
              "homeWorldName": "Tatooine",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ],
          "favorites": undefined,
          "showFavorites": undefined,
          "targetFavorites": undefined,
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
